src/pipeline/rackhd/ova/FunctionTest.groovy

```diff
@@ -0,0 +1,81 @@
+package pipeline.rackhd.ova
+
+def keepEnv(String library_dir, boolean keep_env, int keep_minutes, String test_target, String test_name){
+    def shareMethod = new pipeline.common.ShareMethod()
+    String target_dir = test_target + "/" + test_name + "_$NODE_NAME"
+    if(keep_env){
+        def message = "Job Name: ${env.JOB_NAME} \n" + "Build Full URL: ${env.BUILD_URL} \n" + "Status: FAILURE \n" + "Stage: $test_target/$test_name \n" + "Node Name: $NODE_NAME \n" + "Reserve Duration: $keep_minutes minutes \n"
+        echo "$message"
+        slackSend "$message"
+        sleep time: keep_minutes, unit: 'MINUTES'
+    }
+}
+
+def runTest(String stack_type, String test_name, ArrayList<String> used_resources, Map ova_dict, boolean keep_env_on_failure, int keep_minutes,String use_prebuilt_ova="false"){
```

Comment:
I don't think it's a good idea to pass ```use_prebuilt_ova``` to the function.
The function just need a dictionary "ova_dict" to unstash ova.
If there is a pipeline which allows to use prebuilt ova, there should be a method downloadOVAFromURL and the method return the ova path. The the jenkinsfile call stashFile(ova_path) to get the ova_dict

Reply:
But ovftool support to deploy ova just with url

Reply:
How about replacing use_prebuilt_ova with ova_url?
In this way, the key of the ova_dict is fixed.

Reply:
good idea ~ this can reduce the number of args~